extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -287,40 +317,49 @@ private boolean isPositionalChannelLinked() {
     public void triggerEvent(String channelId, String event) {
         if (getThing().getChannel(channelId) != null) {
             triggerChannel(getThing().getChannel(channelId).getUID(), event);
-        } else {
-            logger.warn("Event {} in thing {} does not exist, please recreate the thing", event, getThing().getUID());
+            return;
```

Comment:
This part is clearly not related to this PR because the topic of it is to replace the scheduler. Next time please create a separate PR for such changes. This makes the review process a lot easier :)

If you want to return early in a method, please do a check first that leads to the return and then do the logic afterwards. Returning in the middle of a method makes it unreadable.

So please change it to
```
//1. check
if (getThing().getChannel(channelId) == null) {
    logger.warn("Event {} in thing {} does not exist, please recreate the thing", event, getThing().getUID());
    return;
}
//2. logic
triggerChannel(getThing().getChannel(channelId).getUID(), event);
```